### DISCUSSION TITLE
How to get JWK set from issuer or client

### DISCUSSION THREAD

**Discussion by grug** [panva/openid-client#558] [Q&A] [answered]
Hi there,

When I initialise an `Issuer` I get a `jwks_uri` which is correct. However, I'd like to be able to access the list of keys at that URI. If I wanted to, I could do a `fetch` on them myself, but I assume that `node-openid-client` has this built in but I just can't find it anywhere. Can someone point me in the right direction?

For some extra context - I want to use one of the JWK public keys to verify an API request that contains a JWT (I've got an API that is consumed by other services - not frontends/humans)

**Comment by panva** [comment, marked as answer]
`openid-client` has an internal cache and representation, internal and not exposed.

> I could do a fetch on them myself

You should. My [recommendation](https://github.com/panva/jose/blob/v4.11.2/docs/functions/jwks_remote.createRemoteJWKSet.md#function-createremotejwkset) for an effortless strategy.

**Reply by grug** [reply]
Thank you for such a fast response! I'll give `jose` a go. Out of curiosity, why isn't it exposed? It doesn't seem uncommon to have an API that is consumed by other services and wanting to verify a JWT against a public cert - or am I missing something here?

**Reply by panva** [reply]
It used to be there. I'm sure I had good enough reason not to carry it over.
